### PR TITLE
ignore the update event if pod is allocated in cache but not present in NodeName

### PR DESCRIPTION
**Pull request by xing0821**
```
I1123 21:11:51.708379       1 session.go:256] After allocated Task <benchmark/priority-low-1060-7761/1a1b0e9e-bc9d-47a1-8cb8-b0f121f8a3ea> to Node <hollow-node-arm-56578c6f86-xm79x>: idle <cpu 750.00, memory 4026531840.00>, used <cpu 250.00, memory 0.00>, releasing <cpu 0.00, memory 0.00>
E1123 21:14:05.510879       1 event_handlers.go:175] task <benchmark/priority-low-1060-7761> receive pod with nodeName  but nodeName is hollow-node-arm-56578c6f86-xm79x in cache
I1123 21:14:08.411237       1 session.go:256] After allocated Task <benchmark/priority-low-1060-7761/1a1b0e9e-bc9d-47a1-8cb8-b0f121f8a3ea> to Node <hollow-node-amd-6bcc678d9-fbc6j>: idle <cpu 750.00, memory 4026531840.00>, used <cpu 250.00, memory 0.00>, releasing <cpu 0.00, memory 0.00>
ERROR: logging before flag.Parse: E1123 21:14:08.688966       1 cache.go:123] Failed to bind pod <benchmark/priority-low-1060-7761>: &errors.StatusError{ErrStatus:v1.Status{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ListMeta:v1.ListMeta{SelfLink:"", ResourceVersion:"", Continue:"", RemainingItemCount:(*int64)(nil)}, Status:"Failure", Message:"Operation cannot be fulfilledon pods/binding \"priority-low-1060-7761\": pod priority-low-1060-7761 is already assigned to node \"hollow-node-arm-56578c6f86-xm79x\"", Reason:"Conflict", Details:(*v1.StatusDetails)(0x4055d8acc0), Code:409}}

```
  the event of pods has been arrived after schedule but without nodeName(because of delay or pressure?), and eventhandler will delete the old task(with allocated status) and put the newone(with pending status) in cache. so the next schedule routine will schedule it again.

### CODE REVIEW

**Comment by Thor-wl** [comment]
Thanks for your contribution! Welcome to Volcano community! Pls fix the DCO first.

File: pkg/scheduler/cache/event_handlers.go [hunk 1]
```diff
@@ -129,8 +129,25 @@ func (sc *SchedulerCache) updateTask(oldTask, newTask *schedulingapi.TaskInfo) e
...
+
+	if job, found := sc.Jobs[pi.Job]; found {
+		if t,found := job.Tasks[pi.UID]; found {
+			return schedulingapi.AllocatedStatus(t.Status)
+		}
+	}
+
+	return false
+}
+
```
Comment (Thor-wl) [inline pkg/scheduler/cache/event_handlers.go:143]: I'm OK about the funciton overall. Sugget add logs and comments for it.

Comment (xing0821) [reply pkg/scheduler/cache/event_handlers.go:144]: Add some log when ignore the update event

**Comment by xing0821** [comment]
> /lgtm

sorry for

**Comment by xing0821** [comment]
@Thor-wl  sorry for waste your time, fmt the code for verify

**Review by Thor-wl — APPROVED** [review]

**Pull request merged**